test/distributed/org/apache/cassandra/distributed/upgrade/UpgradeTestBase.java

```diff
@@ -177,6 +178,10 @@ public void run() throws Throwable
                             cluster.get(n).shutdown().get();
                             cluster.get(n).setVersion(version);
                             cluster.get(n).startup();
+                            // If using in-jvm Dtest networking, update the messaging versions
+                            // so that message filters are able to serialize/deserialize correctly.
+                            if (!cluster.get(n).config().has(Feature.NETWORK))
+                                cluster.updateMessagingVersions();
```

Comment:
Looks like `updateMessagingVersions()` is already called during `cluster.get(n).startup()`. 

```
AbstractCluster.Wrapper#startup(org.apache.cassandra.distributed.api.ICluster)
```

Do we need to invoke it twice?

Reply:
I'd missed that and made this change to make sure the messaging version was set on upgrade. I added the guard around the NETWORK feature so as not to affect any version negotiation attempts if using real networking.

It's probably safest to leave it alone. I'll remove from my code.